src/blog/intl.md

```diff
@@ -0,0 +1,64 @@
+---
+title: 'Faster and more feature-rich internationalization APIs'
+author: '[சத்யா குணசேகரன் (Sathya Gunasekaran)](https://twitter.com/_gsathya)'
+date: 2019-04-25 16:45:37
+avatars:
+  - 'sathya-gunasekaran'
+tags:
+  - ECMAScript
+  - Intl
+---
+The ECMAScript Internationalization API Specification (ECMA-402, or `Intl`) provides key locale-specific functionality such as date formatting, number formatting, plural form selection, and collation. The Chrome V8 and Google Internationalization teams have been collaborating on adding features to V8’s ECMA-402 implementation, while cleaning up technical debt and improving performance and interoperability with other browsers.
+
+## Underlying architectural improvements
+
+Initially the ECMA-402 spec was implemented mostly in JavaScript using V8-extensions and lived outside the V8 codebase. Using the external Extension API meant that several of V8’s internally used APIs for type checking, lifetime management of external C++ objects and internal private data storage couldn’t be used. As part of improving startup performance, this implementation was later moved in to the V8 codebase to enable [snapshotting](/blog/custom-startup-snapshots) of these builtins.
+
+V8 uses specialized `JSObject`s with custom [shapes (hidden classes)](https://mathiasbynens.be/notes/shapes-ics) to describe built-in JavaScript objects specified by ECMAScript (like `Promise`s, `Map`s, `Set`s, etc). With this approach, V8 can pre-allocate the required number of internal slots and generate fast accesses to these, rather than grow the object one property at a time leading to slower performance and worse memory usage.
+
+The `Intl` implementation was not modeled after such an architecture, as a consequence of the historic split. Instead, all the built-in JavaScript objects as specified by the Internationalization spec (like `NumberFormat`, `DateTimeFormat`) were generic `JSObject`s that had to transition through several property additions for their internal slots.
+
+Another artifact of not having a specialized `JSObject`s was that type checking was now more complex. The type information was stored under a private symbol and type-checked on both the JS and C++ side using expensive property access, rather than just looking up its shape.
+
+### Modernizing the codebase
+
+With the current move away from writing self-hosted builtins in V8, it made sense to use this opportunity to modernize the ECMA402 implementation.
```

Comment:
what does "self-hosted" mean?

Reply:
Implemented in JS as part of V8. @gsathya, WDYT? Should we link to https://en.wikipedia.org/wiki/Self-hosting or explain this in a few words?

Reply:
Linking to the wikipedia article sounds good to me.

Reply:
@sam-github Would you like to submit a pull request for this? :)